resource-aggregate/events/resourceLinksSnapshotTaken.go

```diff
@@ -45,7 +45,35 @@ func (e *ResourceLinksSnapshotTaken) IsSnapshot() bool {
 	return true
 }
 
-func (e *ResourceLinksSnapshotTaken) HandleEventResourceLinksPublished(ctx context.Context, pub *ResourceLinksPublished) error {
+// Check if the snapshot of published resources matches exactly the new published data
+func (e *ResourceLinksSnapshotTaken) MatchesLinksPublished(published *ResourceLinksPublished) bool {
+	if published == nil {
+		return false
+	}
+
+	if len(e.Resources) != len(published.Resources) {
```

Comment:
ResourceLinkPublish operation can be by the devices split into multiple publish messages. This condition will automatically consider the received subset as not matching one, even though 3 published resources are already stored and were not changed. Duplicate events are published.